pym/bob/input.py

```diff
@@ -603,6 +603,72 @@ def isDeterministic(self):
     def hasJenkinsPlugin(self):
         return True
 
+
+class CvsScm(BaseScm):
+    # Checkout using CVS
+    # - mandatory parameters: cvsroot, module
+    # - optional parameters: rev, dir (dir is required if there are multiple checkouts)
+    def __init__(self, spec):
+        super().__init__(spec)
+        self.__cvsroot = spec["cvsroot"]
+        self.__module = spec["module"]
+        self.__rev = spec.get("rev")
+        self.__dir = spec.get("dir", ".")
+
+    def resolveEnv(self, env):
+        # Resolve macros in CVS names
+        super().resolveEnv(env)
+        self.__cvsroot = env.substitute(self.__cvsroot, "cvs::cvsroot")
+        self.__module = env.substitute(self.__module, "cvs::module")
+        if self.__rev:
+            self.__rev = env.substitute(self.__rev, "cvs::rev")
+        if self.__dir:
+            self.__dir = env.substitute(self.__dir, "cvs::dir")
+
+    def asScript(self):
+        revarg = "-r {rev}".format(rev=self.__rev) if self.__rev != None else "-A"
+        return """
+# Validate that we're talking to the correct repository. If not, ditch it.
```

Comment:
OK, didn't know about the "attic" feature yet.

I was mainly worrying about users having messed with the workspace directory in unexpected ways. In this case CVS will happily do a ``cvs up`` without knowing that it may not be the module we expect. But then, it's probably a case of "you broke it, keep both halves", and nuking the directory is not the nicest thing to do.